Silicon/TigerlakePkg/Library/Stage1BSocInitLib/Stage1BSocInitLib.c

```diff
@@ -79,18 +79,21 @@ IsWarmReset (
   VOID
   )
 {
-  LOADER_GLOBAL_DATA    *LdrGlobal;
+  VOID                  *FspHobList;
   MEMORY_PLATFORM_DATA  *MemPlatformData;
   UINT8                 ResetType;
   BOOLEAN               IsWarmReset;
 
   IsWarmReset     = FALSE;
-  LdrGlobal       = (LOADER_GLOBAL_DATA *)GetLoaderGlobalDataPointer ();
-  MemPlatformData = GetGuidHobData (
-                      LdrGlobal->FspHobList,

```

Comment:
How about changing LdrGlobal->FspHobList to GetFspHobListPtr () directly?

Reply:
`LdrGlobal->FspHobList` is the removed code. This patch is using GetFspHobListPtr().

Reply:
I mean instead of using new code from line 88 ~ 96, how about we update removed code to only change 1 line.

Reply:
LdrGlobal->FspHobList or GetFspHobListPtr () can be NULL until MemoryInit is done. If the 1st parameter of GetGuidHobData() is NULL, GetGuidHobData will use SblHob instead of FspHob. Better to check if GetFspHobListPtr() is NULL by Caller.